### PR TITLE
Check/CKV_GCP_72

### PR DESCRIPTION
**Pull request by JamesWoolfenden**
4. Ensure Integrity Monitoring for Shielded GKE Nodes is Enabled

This is a default, so nothing is valid as is set explicitly to true but not false
```terraform
shielded_instance_config {
    enable_integrity_monitoring=true
}
```

### CODE REVIEW

File: checkov/terraform/checks/resource/gcp/GKEEnsureIntegrityMonitoring.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            node=conf["node_config"][0]
+            if 'shielded_instance_config' in node.keys():
+                monitor=node["shielded_instance_config"][0]
+                if monitor["enable_integrity_monitoring"] == [True]:
+                    return CheckResult.PASSED
+                else:
+                    return CheckResult.FAILED
+            else:
+                #as default is true this is a pass 
+                return CheckResult.PASSED
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/gcp/GKEEnsureIntegrityMonitoring.py:24]: @JamesWoolfenden Looks like a `BaseResourceNegativeValueCheck` check

Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/gcp/GKEEnsureIntegrityMonitoring.py:24]: but its valid to have no node_config at all

Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/gcp/GKEEnsureIntegrityMonitoring.py:24]: and nothing for shielded_instance_config

Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/gcp/GKEEnsureIntegrityMonitoring.py:24]: unless theres something you need to explain about BaseResourceNegativeValueCheck that i need to know?

Comment (nimrodkor) [reply checkov/terraform/checks/resource/gcp/GKEEnsureIntegrityMonitoring.py:24]: @JamesWoolfenden Let's talk :)

**Review by nimrodkor — APPROVED** [review]

**Pull request merged**